analyses/molecular-subtyping-HGG/03-HGG-molecular-subtyping-cnv.Rmd

```diff
@@ -126,24 +183,37 @@ gistic_status_df <- gistic_df %>%
               . < 0 ~ "loss",
               . > 0 ~ "gain",
               . == 0 ~ "neutral"
-            ))) %>%
-  # drop sample_id
-  select(-sample_id)
+            )))
 
 head(gistic_status_df, n = 10)
 ```
 
+Add in participant IDs.
+
+```{r}
+gistic_status_df <- hgg_metadata_df %>% 
+  select(Kids_First_Biospecimen_ID,
+         Kids_First_Participant_ID,
+         sample_id) %>%
+  right_join(gistic_status_df)
+```
+
 ## Join Together
 
 ```{r}
-# There are some samples that had too many segments and were flagged by GISTIC
-# These values will be filled with NA -- we'll change them to indicate that 
-# they failed GISTIC QC
-final_hgg_cn_df <- left_join(cn_df_genes_status,
-                             gistic_status_df,
-                             by = "Kids_First_Biospecimen_ID") %>%
-  replace(is.na(.), "Failed GISTIC QC")
-head(final_hgg_cn_df, n = 10)
+final_hgg_cn_df <- full_join(cn_df_genes_status,
+                             gistic_status_df)
+
+nrow(final_hgg_cn_df)
+```
+
+`final_hgg_cn_df` includes all the samples that are in the consensus SEG file.
+So any missing CNA values at this stage come from the fact that neutral values are _not_ included in the annotated copy number files.
+We can fill any "missing values" with neutral calls as a result.
```

Comment:
So this is referring to samples that did have CNV data, but didn't have a value for a particular gene because it was neutral?

Reply:
Yes, you are correct. If it's in the GISTIC file, that means it had copy number data and it is not included in the annotated file because neutral calls get filtered out.